code/modules/core_implant/cruciform/upgrades.dm

```diff
@@ -30,3 +30,140 @@
+
+/obj/item/weapon/cruciform_upgrade/proc/OnUninstall()
+
+/obj/item/weapon/cruciform_upgrade/natures_blessing
+	name = "Natures blessing"
+	desc = "This upgrade slowly heals and fertilizes all plants near the follower. Useful for Agrolytes."
+	icon_state = "natures_blessing"
+	matter = list(MATERIAL_BIOMATTER = 100, MATERIAL_GOLD = 5, MATERIAL_PLASTEEL = 5)
+
+/obj/item/weapon/cruciform_upgrade/natures_blessing/OnInstall(var/disciple, var/_cruciform)
+	..()
+	START_PROCESSING(SSobj, src)
+
+/obj/item/weapon/cruciform_upgrade/natures_blessing/OnUninstall()
+	..()
+	STOP_PROCESSING(SSobj, src)
+
+/obj/item/weapon/cruciform_upgrade/natures_blessing/Process()
+	for(var/obj/machinery/portable_atmospherics/hydroponics/tray in oviewers(5, wearer))
+		if(tray.seed && !tray.dead)
+			var/max_health = tray.seed.get_trait(TRAIT_ENDURANCE)
+			if(tray.health < max_health)
+				tray.health += 0.1
+			if(tray.weedlevel)
+				tray.weedlevel -= 0.1
+
+/obj/item/weapon/cruciform_upgrade/faiths_shield
+	name = "Faiths shield"
+	desc = "This upgrade slightly increase follower resistance to Brute and Burn damage from all sources."
+	icon_state = "faiths_shield"
+	matter = list(MATERIAL_BIOMATTER = 50, MATERIAL_GOLD = 5, MATERIAL_PLASTEEL = 10)
+	var/shield_mod = 0.1
+
+/obj/item/weapon/cruciform_upgrade/faiths_shield/OnInstall(var/disciple, var/_cruciform)
+	..()
+	wearer.species.brute_mod -= shield_mod
+	wearer.species.burn_mod -= shield_mod
+
+/obj/item/weapon/cruciform_upgrade/faiths_shield/OnUninstall()
+	..()
+	wearer.species.brute_mod += shield_mod
+	wearer.species.burn_mod += shield_mod
+
+/obj/item/weapon/cruciform_upgrade/cleansing_presence
+	name = "Cleansing presence"
+	desc = "This upgrade cleans tiles that the follower walks upon. Will slowly cause space vines and maint-shrooms to wither and die in the follower’s presence. Useful for Custodians."
+	icon_state = "cleansing_presence"
+	matter = list(MATERIAL_BIOMATTER = 50, MATERIAL_GOLD = 5, MATERIAL_PLASTEEL = 5)
+	var/area_radius = 5
+
+/obj/item/weapon/cruciform_upgrade/cleansing_presence/OnInstall(var/disciple, var/_cruciform)
+	..()
+	START_PROCESSING(SSobj, src)
+
+/obj/item/weapon/cruciform_upgrade/cleansing_presence/OnUninstall()
+	..()
+	STOP_PROCESSING(SSobj, src)
+
+/obj/item/weapon/cruciform_upgrade/cleansing_presence/Process()
+	wearer.clean_blood()
+	var/list/affected = range(area_radius, src)
+	for(var/obj/effect/plant/shroom in affected)
+		if(istype(shroom.seed, /datum/seed/mushroom/maintshroom))
+			shroom.health -= (shroom.max_health * 0.1)
+			shroom.check_health()
+
+/obj/item/weapon/cruciform_upgrade/martyr_gift
+	name = "Martyr gift"
+	desc = "This upgrade upon the death of the follower, causes a large ‘explosion’ that doesn’t damage terrain nor does it damage followers of NT. It will cause massive burns to any non-cruciformed crewmembers or critters within its radius, however; this upgrade will be destroyed upon use, leaving the cruciform un-upgraded when cloned."
```

Comment:
```suggestion
	desc = "This upgrade upon the death of the follower, causes a large ‘explosion’ that doesn’t damage terrain nor does it damage followers of NT. It will cause massive burns to any non-cruciformed crewmembers or critters within its radius. However, in process doing so, this upgrade destroys itself, albeit cruciform remains intact.
```